compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/types/BLangArrayType.java

```diff
@@ -62,10 +65,15 @@ public String toString() {
         final StringBuilder[] sb = {new StringBuilder(getTypeName())};
         if (sizes.length == 0) {
             Arrays.stream(sizes).forEach(size -> {
-                if (size == -1) {
-                    sb[0].append("[]");
+                if (size.getKind() == NodeKind.LITERAL || size.getKind() == NodeKind.NUMERIC_LITERAL) {
```

Comment:
We actually discussed about this before in [this](https://github.com/ballerina-platform/ballerina-lang/pull/25551#discussion_r482686991) and it was suggested to use both. Since it is the same logic that is being used here I used both.

Reply:
IIRC digits are usually represented by `NUMERIC_LITERAL`s, while strings are represented by `LITERAL`. 

So once we've validated that the size is a numeric literal (probably done by the parser itself), we don't have to check for `LITERAL` thereafter. Because we know that if it is an array's size, and it is not inferred or a const ref, it has to be a `NUMERIC_LITERAL`.